### PR TITLE
Get articles topic

### PR DESCRIPTION
**Pull request by AudeNgD**
Complete setup for get articles by topic

### CODE REVIEW

File: __tests__/app.tests.js [hunk 1]
```diff
@@ -230,6 +230,46 @@ describe("app", () => {
...
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.length).toEqual(1);
+          body.forEach((article) => {
+            expect(article).toEqual(
+              expect.objectContaining({
+                article_id: expect.any(Number),
+                title: expect.any(String),
+                topic: expect.any(String),
```
Comment (nc-bela-bertalan) [inline __tests__/app.tests.js:245]: Please check the exact topic here to make sure it's the right one.

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -19,8 +20,16 @@ exports.getArticleById = (req, res, next) => {
...
+
+  const queries = [fetchAllArticles(topic)];
+
+  if (topic) {
+    queries.push(checkTopicExists(topic));
+  }
+  return Promise.all(queries)
+    .then((response) => {
+      const articles = response[0];
       res.status(200).send(articles);
```
Comment (nc-bela-bertalan) [inline controllers/articles.controllers.js:33]: Please send articles back in an object on a key of 'articles'.

**Review by nc-bela-bertalan — COMMENTED** [review]
Hey Aude, please see my comments with instructions for some refactoring.

File: __tests__/app.tests.js (hunk 1 above)
Comment (AudeNgD) [reply __tests__/app.tests.js:245]: Refactored

File: controllers/articles.controllers.js (hunk 2 above)
Comment (AudeNgD) [reply controllers/articles.controllers.js:33]: Refactored

**Pull request merged**